built_in_tasks/target_capture_task.py

```diff
@@ -1062,3 +1062,332 @@ def sequence_2D(nblocks=100, distance=6.8):
+    Saccade task, but subjects need to hold an initial target with their hand
+    '''
+    # TODO make different color targets for hand cursor
+
+    fixation_radius = traits.Float(2.5, desc="Distance from center that is considered a broken fixation")
+    fixation_penalty_time = traits.Float(1.0, desc="Time in fixation penalty state")
+    fixation_target_color = traits.OptionsList("cyan", *target_colors, desc="Color of the eye target under fixation state", bmi3d_input_options=list(target_colors.keys()))
+    eye_target_color = traits.OptionsList("white", *target_colors, desc="Color of the eye target", bmi3d_input_options=list(target_colors.keys()))
+    fixation_radius_buffer = traits.Float(.5, desc="additional radius for eye target")
+
+    status = dict(
+        wait = dict(start_trial="target"),
+        target = dict(leave_target2="hold_penalty",timeout="timeout_penalty",enter_target="hold"),
+        hold = dict(leave_target2="hold_penalty",leave_target="target", gaze_target="fixation"), # must hold an initial hand-target and eye-target
+        fixation = dict(leave_target="delay_penalty",hold_complete="delay", fixation_break="fixation_penalty"), # must hold an initial hand-target and eye-target to initiate a trial
+        delay = dict(leave_target="delay_penalty", delay_complete="targ_transition", fixation_break="fixation_penalty"),
+        targ_transition = dict(trial_complete="reward", trial_abort="wait", trial_incomplete="target"),
+        timeout_penalty = dict(timeout_penalty_end="targ_transition", end_state=True),
+        hold_penalty = dict(hold_penalty_end="targ_transition", end_state=True),
+        delay_penalty = dict(delay_penalty_end="targ_transition", end_state=True),
+        fixation_penalty = dict(fixation_penalty_end="targ_transition",end_state=True),
+        reward = dict(reward_end="wait", stoppable=False, end_state=True)
+    )
+ 
+    sequence_generators = ['row_target','sac_hand_2d']
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Instantiate the targets
+        instantiate_targets = kwargs.pop('instantiate_targets', True)
+        if instantiate_targets:
+
+            # Target 1 and 2 are for saccade. Target 3 is for hand
+            target1 = VirtualCircularTarget(target_radius=self.fixation_radius, target_color=target_colors[self.eye_target_color])
+            target2 = VirtualCircularTarget(target_radius=self.fixation_radius, target_color=target_colors[self.eye_target_color])
+            target3 = VirtualCircularTarget(target_radius=self.target_radius, target_color=target_colors[self.target_color])
+
+            self.targets = [target1, target2]
+            self.targets_hand = [target3]
+
+    # def init(self):
+    #     super().init()
+    #     self.trial_dtype = np.dtype([('trial', 'u4'), ('index', 'u4'), ('target', 'f8', (3,)),('index_hold', 'u4'), ('target_hold', 'f8', (3,))])
+    #     
+    #     super(TargetCapture, self).init()
```

Comment:
why is this commented out / how do you save the hold target position on each trial?